lib/os/crc_shell.c

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2022 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <getopt.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/sys/crc.h>
+
+enum crc_type {
+	CRC16,
+	CRC8,
+	CRC16_CCITT,
+	CRC16_ITU_T,
+	CRC16_ANSI,
+	CRC32_IEEE,
+	CRC32_C,
+	CRC8_CCITT,
+	CRC7_BE,
+};
+
+/* These must match the order of enum crc_type */
+static const char *const crc_types[] = {
+	"16", "8", "16_ccitt", "16_itu_t", "16_ansi", "32_ieee", "32_c", "8_ccitt", "7_be",
+};
```

Comment:
This could be slightly error prone if one adds new crc enum values but misses the corresponding string.
We could do it a bit more safely with this
```
static const char *const crc_types[] = {
	[CRC8] = "8",
	[CRC16] = "16",
...
};
```

Reply:
Done!